src/kubernetes_api_objects/pod.rs

```diff
@@ -1081,6 +1172,46 @@ impl Marshalable for ContainerView {
     proof fn marshal_preserves_integrity() {}
 }
 
+pub struct LifecycleView {
+    pub pre_stop: Option<LifecycleHandlerView>,
+}
+
+impl LifecycleView {
+    pub open spec fn default() -> LifecycleView {
+        LifecycleView {
+            pre_stop: Option::None,
+        }
+    }
+
+    pub open spec fn set_pre_stop(self, pre_stop: LifecycleHandlerView) -> LifecycleView {
+        LifecycleView {
+            pre_stop: Option::Some(pre_stop),
+            ..self
+        }
+    }
+
+}
+
+
+pub struct LifecycleHandlerView {
+    pub execs: Option<Seq<StringView>>,
+}
+
+impl LifecycleHandlerView {
+    pub open spec fn default() -> LifecycleHandlerView {
+        LifecycleHandlerView {
+            execs: Option::None,
+        }
+    }
+
+    pub open spec fn set_exec(self, commands: Seq<StringView>) -> LifecycleHandlerView {
```

Comment:
Better to name the argument consistent with the field, that is `exec`

Reply:
Currently we only support to use commands in exec. I think we can leave it as 'commands' here and if we extend the exec later, we can modify the name.